astropy/io/fits/src/compressionmodule.c

```diff
@@ -1114,10 +1116,12 @@ void compression_module_init(PyObject* module) {
     cfitsio_version = floor((1000 * version_tmp + 0.5)) / 1000;
```

Comment:
Does anyone know why this is here? The explanation + that handling was added in https://github.com/spacetelescope/PyFITS/commit/9ebaf57a152c4e16867d5e9577820d0e0d7afa4a but I really cannot think of a case where `float` -> `double` could loose precision...

Reply:
cc: @embray 🙏

Reply:
According to the C99 standard (section 6.3.1.5:1):
> When a float is promoted to double or long double, or a double is promoted
to long double, its value is unchanged (if the source value is represented in the
precision and range of its type).

So I'm not really sure either, but maybe it's best to leave it alone for now.